Core/Executor/ContentManager.php

```diff
@@ -595,33 +595,137 @@ public function generateMigration(array $matchCondition, $mode, array $context =
+        }
+    }
+
+    /**
+     * @param $fields
+     * @return array
+     */
+    private function convertFields($fields)
+    {
+        $convertedFields = [];
         $i = 0;
         // the 'easy' yml: key = field name, value = value
         // deprecated: the 'legacy' yml: key = numerical index, value = array ( field name => value )
         foreach ($fields as $key => $field) {
-
             if ($key === $i && is_array($field) && count($field) == 1) {
                 // each $field is one key value pair
                 // eg.: $field = array($fieldIdentifier => $fieldValue)
                 reset($field);
                 $fieldIdentifier = key($field);
                 $fieldValue = $field[$fieldIdentifier];
+
+                $convertedFields[$fieldIdentifier] = $fieldValue;
             } else {
-                $fieldIdentifier = $key;
-                $fieldValue = $field;
+                $convertedFields[$key] = $field;
             }
+            $i++;
+        }
 
-            if (!isset($contentType->fieldDefinitionsByIdentifier[$fieldIdentifier])) {
-                throw new \Exception("Field '$fieldIdentifier' is not present in content type '{$contentType->identifier}'");
+        return $convertedFields;
+    }
+
+    /**
+     * @param array $fields
+     * @param string $language
+     * @return array
+     */
+    protected function parseSingleLangFields(array $fields, $language)
+    {
+        $fieldsList = [];
+        foreach ($fields as $fieldIdentifier => $fieldValue) {
+            $fieldsList[] = [$fieldIdentifier, $fieldValue, $language];
+        }
+
+        return $fieldsList;
+    }
+
+    /**
+     * @param array $fields
+     * @return array
+     * @throws \Exception
+     */
+    protected function parseMultiLangFields(array$fields)
+    {
+        $fieldsList = [];
+
+        foreach ($fields as $key => $field) {
+            if (!is_array($field)) {
+                throw new \Exception("Field '$key' should use multilang syntax as other fields in this step");
             }
 
-            $fieldDefinition = $contentType->fieldDefinitionsByIdentifier[$fieldIdentifier];
-            $fieldValue = $this->getFieldValue($fieldValue, $fieldDefinition, $contentType->identifier, $step->context);
+            foreach($field as $languageCode => $value) {
+                $fieldsList[] = [$key, $value, $languageCode];
+            }
+        }
 
-            $createOrUpdateStruct->setField($fieldIdentifier, $fieldValue, $this->getLanguageCode($step));
+        return $fieldsList;
+    }
 
-            $i++;
+    /**
+     * Checks whatever at least one of field is using multilang syntax.
+     *
+     * @param array $fields
+     * @return bool
+     */
+    protected function hasLanguageCodesAsKeys(array $fields)
+    {
+        $languages = $this->getContentLanguages();
+        $languageCodes = array_combine($languages, $languages);
+        $hasLanguageCodesAsKeys = false;
```

Comment:
I'd rather do the opposite logic: start the function with True and exit with False as soon as one non-multilang field def (or multilang with wrong key) is detected.
Will merge as is, then fixup